vpr/src/route/rr_metadata.cpp

```diff
@@ -7,33 +7,33 @@ namespace vpr {
 const t_metadata_value* rr_node_metadata(int src_node, vtr::interned_string key) {
     auto& device_ctx = g_vpr_ctx.device();
 
-    auto iter = device_ctx.rr_node_metadata.find(src_node);
-    if (iter == device_ctx.rr_node_metadata.end()) {
+    auto iter = device_ctx.rr_graph_builder.rr_node_metadata_find(src_node);
+    if (iter == device_ctx.rr_graph_builder.rr_node_metadata_end(src_node)) {
         return nullptr;
     }
     return iter->second.one(key);
 }
 
 void add_rr_node_metadata(int src_node, vtr::interned_string key, vtr::interned_string value) {
     auto& device_ctx = g_vpr_ctx.mutable_device();
-    device_ctx.rr_node_metadata.add_metadata(src_node,
-                                             key,
-                                             value);
+    device_ctx.rr_graph_builder.rr_node_metadata_add_metadata(src_node,
```

Comment:
In principle, we should not reinvent the wheel. The ideal way is 

```
device_ctx.rr_graph_builder.rr_node_metadata().add_metadata(..)
```

I remember you mentioned it caused some errors in compilation. If so, I am o.k. for now.

Reply:
The API rr_node_metadata_add_metadata has been removed and the add_metadata has been called as the prescribed method.